src/main/scala/edu/ie3/simona/model/participant2/ParticipantModelShell.scala

```diff
@@ -199,6 +199,12 @@ final case class ParticipantModelShell[
     def modelOperatingPoint(): (OP, OperationChangeIndicator) = {
       val (modelOp, modelNextTick) =
         model.determineOperatingPoint(currentState)
+      // Sanity check
+      if (modelNextTick.contains(tick))
+        throw new CriticalFailureException(
+          s"Next tick ($modelNextTick) is same as the current tick ($tick)."
+        )
```

Comment:
We should also check for earlier ticks, who knows what those pesky models would do ;)

```suggestion
      if (modelNextTick.exists(_ <= tick))
        throw new CriticalFailureException(
          s"Next tick ($modelNextTick) is same as or earlier than the current tick ($tick)."
        )
```